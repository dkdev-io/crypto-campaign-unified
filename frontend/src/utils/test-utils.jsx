import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { vi } from 'vitest';

// Test data factories
export const createMockCampaign = (overrides = {}) => ({
  id: 'test-campaign-1',
  campaign_name: 'Test Campaign',
  email: 'test@example.com',
  candidate_name: 'John Doe',
  theme_color: '#2a2a72',
  suggested_amounts: [25, 50, 100, 250],
  max_donation_limit: 3300,
  website: 'https://test.com',
  wallet_address: '0x123...',
  created_at: '2024-01-01T00:00:00.000Z',
  ...overrides
});

export const createMockDonorData = (overrides = {}) => ({
  fullName: 'Jane Smith',
  email: 'jane@example.com',
  phone: '555-123-4567',
  street: '123 Main St',
  city: 'Anytown',
  state: 'CA',
  zip: '12345',
  employer: 'Test Corp',
  occupation: 'Engineer',
  amount: '100',
  ...overrides
});

export const createMockWalletInfo = (overrides = {}) => ({
  isConnected: true,
  account: '0x742d35Cc6C4C5e5D0E9dF5F9B6A5e8a2c1B9D8e7',
  balance: '1.5678',
  network: {
    name: 'Ethereum Mainnet',
    chainId: '0x1'
  },
  contributorInfo: {
    isKYCVerified: true,
    cumulativeAmount: 0.5,
    remainingCapacity: 1.0
  },
  ...overrides
});

export const createMockFormSubmission = (overrides = {}) => ({
  id: 'submission-1',
  campaign_id: 'test-campaign-1',
  donor_full_name: 'Jane Smith',
  donor_email: 'jane@example.com',
  amount_usd: 100,
  transaction_hash: '0xabc123...',
  status: 'confirmed',
  created_at: '2024-01-01T12:00:00.000Z',
  ...overrides
});

// Custom render function with providers
export const render = (ui, {
  route = '/',
  ...renderOptions
} = {}) => {
  // Set initial URL for router
  window.history.pushState({}, 'Test page', route);
  
  const Wrapper = ({ children }) => (
    <BrowserRouter>
      {children}
    </BrowserRouter>
  );

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// Utility to wait for async operations
export const waitForAsync = () => new Promise(resolve => setTimeout(resolve, 0));

// Mock event creators
export const createMockEvent = (overrides = {}) => ({
  preventDefault: vi.fn(),
  stopPropagation: vi.fn(),
  target: {
    value: 'test-value',
    name: 'test-input',
    checked: false,
    ...overrides.target
  },
  currentTarget: {
    value: 'test-value',
    ...overrides.currentTarget
  },
  ...overrides
});

export const createChangeEvent = (value, name = 'test-input') => 
  createMockEvent({
    target: { value, name },
    currentTarget: { value }
  });

export const createClickEvent = (overrides = {}) =>
  createMockEvent({
    type: 'click',
    ...overrides
  });

export const createSubmitEvent = (overrides = {}) =>
  createMockEvent({
    type: 'submit',
    ...overrides
  });

// Form validation helpers
export const fillFormFields = async (user, formData) => {
  const fields = Object.entries(formData);
  
  for (const [fieldName, value] of fields) {
    const input = document.querySelector(`input[name="${fieldName}"], select[name="${fieldName}"], textarea[name="${fieldName}"]`);
    if (input) {
      if (input.type === 'checkbox' || input.type === 'radio') {
        if (value) {
          await user.click(input);
        }
      } else {
        await user.type(input, String(value));
      }
    }
  }
};

// Assertion helpers
export const expectFormField = (container, fieldName, expectedValue = null) => {
  const field = container.querySelector(`[name="${fieldName}"]`);
  expect(field).toBeInTheDocument();
  
  if (expectedValue !== null) {
    if (field.type === 'checkbox' || field.type === 'radio') {
      expect(field.checked).toBe(Boolean(expectedValue));
    } else {
      expect(field.value).toBe(String(expectedValue));
    }
  }
  
  return field;
};

export const expectErrorMessage = (container, message) => {
  const errorElement = container.querySelector('[style*="background: #f8d7da"], [style*="color: #721c24"], .error-message');
  expect(errorElement).toBeInTheDocument();
  if (message) {
    expect(errorElement).toHaveTextContent(message);
  }
};

export const expectSuccessMessage = (container, message) => {
  const successElement = container.querySelector('[style*="background: #d4edda"], [style*="color: #155724"], .success-message');
  expect(successElement).toBeInTheDocument();
  if (message) {
    expect(successElement).toHaveTextContent(message);
  }
};

// Mock HTTP responses
export const mockApiResponse = (data, options = {}) => {
  const {
    status = 200,
    ok = true,
    delay = 0
  } = options;
  
  return new Promise(resolve => {
    setTimeout(() => {
      resolve({
        ok,
        status,
        statusText: ok ? 'OK' : 'Error',
        json: () => Promise.resolve(data),
        text: () => Promise.resolve(JSON.stringify(data))
      });
    }, delay);
  });
};

// Component test helpers
export const getByDataTestId = (container, testId) => {
  return container.querySelector(`[data-testid="${testId}"]`);
};

export const getAllByDataTestId = (container, testId) => {
  return container.querySelectorAll(`[data-testid="${testId}"]`);
};

// Form interaction patterns
export const submitForm = async (user, form) => {
  const submitButton = form.querySelector('button[type="submit"], input[type="submit"]');
  if (submitButton) {
    await user.click(submitButton);
  } else {
    // Submit via form if no submit button found
    await user.submit(form);
  }
};

export const selectSuggestedAmount = async (user, container, amount) => {
  const amountButton = container.querySelector(`button[type="button"]:has-text("$${amount}")`);
  if (amountButton) {
    await user.click(amountButton);
  }
  return amountButton;
};

// Wallet interaction helpers
export const connectWallet = async (user, container) => {
  const connectButton = container.querySelector('button:has-text("Connect"), button:has-text("MetaMask")');
  if (connectButton) {
    await user.click(connectButton);
  }
  return connectButton;
};

export const disconnectWallet = async (user, container) => {
  const disconnectButton = container.querySelector('button:has-text("Disconnect")');
  if (disconnectButton) {
    await user.click(disconnectButton);
  }
  return disconnectButton;
};

// Time utilities for testing
export const mockDateNow = (timestamp) => {
  const dateSpy = vi.spyOn(Date, 'now');
  dateSpy.mockReturnValue(timestamp);
  return dateSpy;
};

export const mockDate = (dateString) => {
  const mockDate = new Date(dateString);
  const dateSpy = vi.spyOn(global, 'Date');
  dateSpy.mockImplementation(() => mockDate);
  return dateSpy;
};

// Export everything for easy importing
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { vi } from 'vitest';