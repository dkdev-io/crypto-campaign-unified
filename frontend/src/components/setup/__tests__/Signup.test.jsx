import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render, screen, userEvent, waitFor } from '../../../utils/test-utils';
import Signup from '../Signup';

describe('Signup', () => {
  const defaultProps = {
    formData: {},
    updateFormData: vi.fn(),
    onNext: vi.fn(),
    onPrev: vi.fn()
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('Component Rendering', () => {
    it('renders signup step with title and description', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByText(/Campaign Setup - Step 1/)).toBeInTheDocument();
      expect(screen.getByText(/Tell us about yourself and your campaign/)).toBeInTheDocument();
    });

    it('renders all required form fields', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByLabelText(/Your Full Name/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Your Email Address/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Campaign Name/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Committee Name or Search Term/)).toBeInTheDocument();
    });

    it('renders optional fields', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByLabelText(/Suggested Contribution Amounts/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Maximum Contribution Limit/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Campaign Website/)).toBeInTheDocument();
    });

    it('renders next button with correct text', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByText('Next: Find Your Committee â†’')).toBeInTheDocument();
    });

    it('does not render back button on first step', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.queryByText(/Back/)).not.toBeInTheDocument();
      expect(screen.queryByText(/Previous/)).not.toBeInTheDocument();
    });
  });

  describe('Form Data Population', () => {
    it('displays existing data from formData props', () => {
      const propsWithData = {
        ...defaultProps,
        formData: {
          userFullName: 'John Doe',
          campaignName: 'Doe for Mayor',
          email: 'john@doe2024.com',
          website: 'https://doe2024.com'
        }
      };

      render(<Signup {...propsWithData} />);
      
      expect(screen.getByDisplayValue('John Doe')).toBeInTheDocument();
      expect(screen.getByDisplayValue('Doe for Mayor')).toBeInTheDocument();
      expect(screen.getByDisplayValue('john@doe2024.com')).toBeInTheDocument();
      expect(screen.getByDisplayValue('https://doe2024.com')).toBeInTheDocument();
    });

    it('shows empty fields when no form data provided', () => {
      render(<Signup {...defaultProps} />);
      
      const nameInput = screen.getByLabelText(/Your Full Name/);
      const campaignInput = screen.getByLabelText(/Campaign Name/);
      const emailInput = screen.getByLabelText(/Email Address/);
      const websiteInput = screen.getByLabelText(/Campaign Website/);
      
      expect(nameInput.value).toBe('');
      expect(campaignInput.value).toBe('');
      expect(emailInput.value).toBe('');
      expect(websiteInput.value).toBe('');
    });
  });

  describe('User Input Handling', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
    });

    it('updates form data when full name is changed', async () => {
      render(<Signup {...defaultProps} />);
      
      const nameInput = screen.getByLabelText(/Your Full Name/);
      await user.type(nameInput, 'Jane Smith');
      
      // Check that updateFormData was called with the final character
      expect(defaultProps.updateFormData).toHaveBeenLastCalledWith({
        userFullName: 'h'
      });
      // Check that it was called the correct number of times
      expect(defaultProps.updateFormData).toHaveBeenCalledTimes(10);
    });

    it('updates form data when campaign name is changed', async () => {
      render(<Signup {...defaultProps} />);
      
      const campaignInput = screen.getByLabelText(/Campaign Name/);
      await user.type(campaignInput, 'Smith for Senate');
      
      // Check that updateFormData was called with the final character
      expect(defaultProps.updateFormData).toHaveBeenLastCalledWith({
        campaignName: 'e'
      });
      // Check that it was called the correct number of times (16 characters)
      expect(defaultProps.updateFormData).toHaveBeenCalledTimes(16);
    });

    it('updates form data when email is changed', async () => {
      render(<Signup {...defaultProps} />);
      
      const emailInput = screen.getByLabelText(/Email Address/);
      await user.type(emailInput, 'jane@smith2024.com');
      
      // Check that updateFormData was called with the final character
      expect(defaultProps.updateFormData).toHaveBeenLastCalledWith({
        email: 'm'
      });
      // Check that it was called the correct number of times (18 characters)
      expect(defaultProps.updateFormData).toHaveBeenCalledTimes(18);
    });

    it('updates form data when website is changed', async () => {
      render(<Signup {...defaultProps} />);
      
      const websiteInput = screen.getByLabelText(/Campaign Website/);
      await user.type(websiteInput, 'https://smith2024.com');
      
      // Check that updateFormData was called with the final character
      expect(defaultProps.updateFormData).toHaveBeenLastCalledWith({
        website: 'm'
      });
      // Check that it was called the correct number of times (21 characters)
      expect(defaultProps.updateFormData).toHaveBeenCalledTimes(21);
    });

    it('calls updateFormData for each character typed', async () => {
      render(<Signup {...defaultProps} />);
      
      const nameInput = screen.getByLabelText(/Your Full Name/);
      await user.type(nameInput, 'Test');
      
      // Should be called 4 times, once for each character
      expect(defaultProps.updateFormData).toHaveBeenCalledTimes(4);
      expect(defaultProps.updateFormData).toHaveBeenNthCalledWith(1, { userFullName: 'T' });
      expect(defaultProps.updateFormData).toHaveBeenNthCalledWith(4, { userFullName: 't' });
    });
  });

  describe('Navigation', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
    });

    it('calls onNext when next button is clicked with valid form', async () => {
      const propsWithValidData = {
        ...defaultProps,
        formData: {
          userFullName: 'John Doe',
          email: 'john@test.com',
          campaignName: 'Test Campaign',
          committeeNameSearch: 'Test Committee'
        }
      };
      render(<Signup {...propsWithValidData} />);
      
      const nextButton = screen.getByText('Next: Find Your Committee â†’');
      await user.click(nextButton);
      
      expect(defaultProps.onNext).toHaveBeenCalledTimes(1);
    });

    it('prevents navigation with empty required fields', async () => {
      render(<Signup {...defaultProps} />);
      
      // Should NOT be able to proceed with empty required fields
      const nextButton = screen.getByText('Next: Find Your Committee â†’');
      await user.click(nextButton);
      
      expect(defaultProps.onNext).not.toHaveBeenCalled();
    });

    it('shows validation errors for empty required fields', async () => {
      render(<Signup {...defaultProps} />);
      
      const nextButton = screen.getByText('Next: Find Your Committee â†’');
      await user.click(nextButton);
      
      await waitFor(() => {
        expect(screen.getByText('Your full name is required')).toBeInTheDocument();
        expect(screen.getByText('Email address is required')).toBeInTheDocument();
        expect(screen.getByText('Campaign name is required')).toBeInTheDocument();
        expect(screen.getByText('Committee name is required')).toBeInTheDocument();
      });
    });
  });

  describe('Form Field Validation', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
    });

    it('email field has email input type', () => {
      render(<Signup {...defaultProps} />);
      
      const emailInput = screen.getByLabelText(/Email Address/);
      expect(emailInput).toHaveAttribute('type', 'email');
    });

    it('website field has url input type', () => {
      render(<Signup {...defaultProps} />);
      
      const websiteInput = screen.getByLabelText(/Campaign Website/);
      expect(websiteInput).toHaveAttribute('type', 'url');
    });

    it('required fields have required attribute', () => {
      render(<Signup {...defaultProps} />);
      
      const nameInput = screen.getByLabelText(/Your Full Name/);
      const campaignInput = screen.getByLabelText(/Campaign Name/);
      const emailInput = screen.getByLabelText(/Email Address/);
      const committeeInput = screen.getByLabelText(/Committee Name or Search Term/);
      const websiteInput = screen.getByLabelText(/Campaign Website/);
      
      expect(nameInput).toBeRequired();
      expect(campaignInput).toBeRequired();
      expect(emailInput).toBeRequired();
      expect(committeeInput).toBeRequired();
      expect(websiteInput).toBeRequired();
    });

    it('shows placeholder text for guidance', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByPlaceholderText('Enter your full name')).toBeInTheDocument();
      expect(screen.getByPlaceholderText('e.g., Smith for Mayor 2024')).toBeInTheDocument();
      expect(screen.getByPlaceholderText('your.email@domain.com')).toBeInTheDocument();
      expect(screen.getByPlaceholderText('https://yourcampaign.com')).toBeInTheDocument();
    });
  });

  describe('Information Sections', () => {
    it('displays what happens next section', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByText('ðŸ”„ What Happens Next')).toBeInTheDocument();
      expect(screen.getByText(/Find Your FEC Committee/)).toBeInTheDocument();
      expect(screen.getByText(/Connect Bank Account/)).toBeInTheDocument();
      expect(screen.getByText(/Launch Your Form/)).toBeInTheDocument();
    });

    it('displays privacy and security information', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByText('ðŸ”’ Privacy & Security')).toBeInTheDocument();
      expect(screen.getByText(/Your information is encrypted/)).toBeInTheDocument();
      expect(screen.getByText(/FEC-compliant data handling/)).toBeInTheDocument();
      expect(screen.getByText(/You control your campaign data/)).toBeInTheDocument();
    });

    it('displays requirements information', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByText('ðŸ“‹ Requirements')).toBeInTheDocument();
      expect(screen.getByText(/Valid FEC committee/)).toBeInTheDocument();
      expect(screen.getByText(/Campaign bank account/)).toBeInTheDocument();
      expect(screen.getByText(/Official campaign website/)).toBeInTheDocument();
    });
  });

  describe('CSS Styling', () => {
    it('applies correct CSS classes to form elements', () => {
      render(<Signup {...defaultProps} />);
      
      const nameInput = screen.getByLabelText(/Your Full Name/);
      const emailInput = screen.getByLabelText(/Email Address/);
      
      expect(nameInput).toHaveClass('form-input');
      expect(emailInput).toHaveClass('form-input');
    });

    it('applies correct styling to containers and sections', () => {
      const { container } = render(<Signup {...defaultProps} />);
      
      expect(container.querySelector('.form-group')).toBeInTheDocument();
      expect(container.querySelector('.form-actions')).toBeInTheDocument();
    });

    it('styles next button with primary class', () => {
      render(<Signup {...defaultProps} />);
      
      const nextButton = screen.getByText('Next: Find Your Committee â†’');
      expect(nextButton).toHaveClass('btn', 'btn-primary');
    });
  });

  describe('Help Text and Instructions', () => {
    it('displays help text for each field', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByText(/This will be used for FEC reporting/)).toBeInTheDocument();
      expect(screen.getByText(/Publicly displayed name for your campaign/)).toBeInTheDocument();
      expect(screen.getByText(/Used for notifications and account management/)).toBeInTheDocument();
      expect(screen.getByText(/Your official campaign website/)).toBeInTheDocument();
    });

    it('provides clear instructions for getting started', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByText(/We'll guide you through the complete setup/)).toBeInTheDocument();
      expect(screen.getByText(/This process takes about 5-10 minutes/)).toBeInTheDocument();
    });
  });

  describe('Special Character Handling', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
    });

    it('handles special characters in name field', async () => {
      const propsWithData = {
        ...defaultProps,
        formData: { userFullName: 'JosÃ© MarÃ­a O\'Connor-Smith' }
      };
      render(<Signup {...propsWithData} />);
      
      const nameInput = screen.getByLabelText(/Your Full Name/);
      expect(nameInput.value).toBe('JosÃ© MarÃ­a O\'Connor-Smith');
    });

    it('handles campaign names with special characters', async () => {
      const propsWithData = {
        ...defaultProps,
        formData: { campaignName: 'Citizens for Change & Progress 2024!' }
      };
      render(<Signup {...propsWithData} />);
      
      const campaignInput = screen.getByLabelText(/Campaign Name/);
      expect(campaignInput.value).toBe('Citizens for Change & Progress 2024!');
    });

    it('handles international domain names in website field', async () => {
      const propsWithData = {
        ...defaultProps,
        formData: { website: 'https://campaÃ±a2024.com' }
      };
      render(<Signup {...propsWithData} />);
      
      const websiteInput = screen.getByLabelText(/Campaign Website/);
      expect(websiteInput.value).toBe('https://campaÃ±a2024.com');
    });
  });

  describe('Accessibility', () => {
    it('has proper heading structure', () => {
      render(<Signup {...defaultProps} />);
      
      const mainHeading = screen.getByRole('heading', { level: 2 });
      expect(mainHeading).toHaveTextContent(/Campaign Setup - Step 1/);
    });

    it('has proper form labels associated with inputs', () => {
      render(<Signup {...defaultProps} />);
      
      expect(screen.getByLabelText(/Your Full Name/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Campaign Name/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Email Address/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Campaign Website/)).toBeInTheDocument();
    });

    it('has accessible button with descriptive text', () => {
      render(<Signup {...defaultProps} />);
      
      const nextButton = screen.getByRole('button', { name: /Next: Find Your Committee/ });
      expect(nextButton).toBeInTheDocument();
    });

    it('maintains focus when typing in inputs', async () => {
      const user = userEvent.setup();
      render(<Signup {...defaultProps} />);
      
      const nameInput = screen.getByLabelText(/Your Full Name/);
      await user.click(nameInput);
      await user.type(nameInput, 'Test Name');
      
      expect(nameInput).toHaveFocus();
    });

    it('has proper ARIA attributes for required fields', () => {
      render(<Signup {...defaultProps} />);
      
      const requiredInputs = [
        screen.getByLabelText(/Your Full Name/),
        screen.getByLabelText(/Campaign Name/),
        screen.getByLabelText(/Email Address/),
        screen.getByLabelText(/Committee Name or Search Term/),
        screen.getByLabelText(/Campaign Website/)
      ];
      
      requiredInputs.forEach(input => {
        expect(input).toHaveAttribute('aria-required', 'true');
      });
    });
  });

  describe('Component Snapshots', () => {
    it('matches snapshot for empty form', () => {
      const { container } = render(<Signup {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for populated form', () => {
      const propsWithData = {
        ...defaultProps,
        formData: {
          userFullName: 'Jane Doe',
          campaignName: 'Doe for Governor 2024',
          email: 'jane@doe2024.org',
          website: 'https://jane-doe-2024.org'
        }
      };

      const { container } = render(<Signup {...propsWithData} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});