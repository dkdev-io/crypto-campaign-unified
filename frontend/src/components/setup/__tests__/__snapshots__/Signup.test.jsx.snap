// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Signup > Component Snapshots > matches snapshot for empty form 1`] = `
<div>
  <h2
    style="color: rgb(42, 42, 114); text-align: center; margin-bottom: 1rem;"
  >
    ðŸ“‹ Campaign Setup - Step 1
  </h2>
  <p
    style="text-align: center; color: rgb(102, 102, 102); margin-bottom: 2rem;"
  >
    Tell us about yourself and your campaign
  </p>
  <div
    class="form-group"
  >
    <label
      for="userFullName"
    >
      Your Full Name *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="userFullName"
      placeholder="Enter your full name"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="text"
      value=""
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      This will be used for FEC reporting
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="email"
    >
      Your Email Address *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="email"
      placeholder="your.email@domain.com"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="email"
      value=""
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Used for notifications and account management
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="campaignName"
    >
      Campaign Name *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="campaignName"
      placeholder="e.g., Smith for Mayor 2024"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="text"
      value=""
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Publicly displayed name for your campaign
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="committeeNameSearch"
    >
      Committee Name or Search Term *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="committeeNameSearch"
      placeholder="Friends of John Smith Committee"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="text"
      value=""
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Enter your FEC committee name or keywords to search. We'll help you find the exact match in the next step.
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="suggestedAmounts"
    >
      Suggested Contribution Amounts
    </label>
    <input
      class="form-input"
      id="suggestedAmounts"
      placeholder="25, 50, 100, 250"
      type="text"
      value="25, 50, 100, 250"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Comma-separated dollar amounts that contributors can quickly select
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="maxDonationLimit"
    >
      Maximum Contribution Limit
    </label>
    <input
      class="form-input"
      id="maxDonationLimit"
      max="10000"
      min="1"
      placeholder="3300"
      type="number"
      value="3300"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Maximum amount a single contributor can donate (FEC limit: $3,300 per election)
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="website"
    >
      Campaign Website (Optional)
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="website"
      placeholder="https://yourcampaign.com"
      required=""
      type="url"
      value=""
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Your official campaign website
    </small>
  </div>
  <div
    class="form-actions"
  >
    <div />
    <button
      class="btn btn-primary"
    >
      Next: Find Your Committee â†’
    </button>
  </div>
  <div
    style="margin-top: 2rem; padding: 1rem; background: rgb(248, 249, 250); border: 1px solid rgb(233, 236, 239); border-radius: 6px; font-size: 14px; color: rgb(108, 117, 125);"
  >
    <h3>
      ðŸ”„ What Happens Next
    </h3>
    <ul
      style="margin: 0.5rem 0px 0px 0px; padding-left: 1.2rem;"
    >
      <li>
        <strong>
          Find Your FEC Committee:
        </strong>
         We'll search for your registered committee
      </li>
      <li>
        <strong>
          Connect Bank Account:
        </strong>
         Secure connection via Plaid
      </li>
      <li>
        <strong>
          Launch Your Form:
        </strong>
         Get your embed code and start collecting
      </li>
    </ul>
    <h3
      style="margin-top: 1rem;"
    >
      ðŸ”’ Privacy & Security
    </h3>
    <ul
      style="margin: 0.5rem 0px 0px 0px; padding-left: 1.2rem;"
    >
      <li>
        <strong>
          Your information is encrypted
        </strong>
         and stored securely
      </li>
      <li>
        <strong>
          FEC-compliant data handling
        </strong>
         with audit trails
      </li>
      <li>
        <strong>
          You control your campaign data
        </strong>
         - export anytime
      </li>
    </ul>
    <h3
      style="margin-top: 1rem;"
    >
      ðŸ“‹ Requirements
    </h3>
    <ul
      style="margin: 0.5rem 0px 0px 0px; padding-left: 1.2rem;"
    >
      <li>
        <strong>
          Valid FEC committee
        </strong>
         registration required
      </li>
      <li>
        <strong>
          Campaign bank account
        </strong>
         for processing donations
      </li>
      <li>
        <strong>
          Official campaign website
        </strong>
         (recommended)
      </li>
    </ul>
    <p
      style="margin-top: 1rem; text-align: center; color: rgb(73, 80, 87);"
    >
      <strong>
        We'll guide you through the complete setup process.
      </strong>
      <br />
      This process takes about 5-10 minutes to complete.
    </p>
  </div>
</div>
`;

exports[`Signup > Component Snapshots > matches snapshot for populated form 1`] = `
<div>
  <h2
    style="color: rgb(42, 42, 114); text-align: center; margin-bottom: 1rem;"
  >
    ðŸ“‹ Campaign Setup - Step 1
  </h2>
  <p
    style="text-align: center; color: rgb(102, 102, 102); margin-bottom: 2rem;"
  >
    Tell us about yourself and your campaign
  </p>
  <div
    class="form-group"
  >
    <label
      for="userFullName"
    >
      Your Full Name *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="userFullName"
      placeholder="Enter your full name"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="text"
      value="Jane Doe"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      This will be used for FEC reporting
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="email"
    >
      Your Email Address *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="email"
      placeholder="your.email@domain.com"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="email"
      value="jane@doe2024.org"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Used for notifications and account management
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="campaignName"
    >
      Campaign Name *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="campaignName"
      placeholder="e.g., Smith for Mayor 2024"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="text"
      value="Doe for Governor 2024"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Publicly displayed name for your campaign
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="committeeNameSearch"
    >
      Committee Name or Search Term *
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="committeeNameSearch"
      placeholder="Friends of John Smith Committee"
      required=""
      style="border-color: rgb(206, 212, 218);"
      type="text"
      value=""
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Enter your FEC committee name or keywords to search. We'll help you find the exact match in the next step.
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="suggestedAmounts"
    >
      Suggested Contribution Amounts
    </label>
    <input
      class="form-input"
      id="suggestedAmounts"
      placeholder="25, 50, 100, 250"
      type="text"
      value="25, 50, 100, 250"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Comma-separated dollar amounts that contributors can quickly select
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="maxDonationLimit"
    >
      Maximum Contribution Limit
    </label>
    <input
      class="form-input"
      id="maxDonationLimit"
      max="10000"
      min="1"
      placeholder="3300"
      type="number"
      value="3300"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Maximum amount a single contributor can donate (FEC limit: $3,300 per election)
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      for="website"
    >
      Campaign Website (Optional)
    </label>
    <input
      aria-required="true"
      class="form-input"
      id="website"
      placeholder="https://yourcampaign.com"
      required=""
      type="url"
      value="https://jane-doe-2024.org"
    />
    <small
      class="field-help"
      style="color: rgb(108, 117, 125); font-size: 12px;"
    >
      Your official campaign website
    </small>
  </div>
  <div
    class="form-actions"
  >
    <div />
    <button
      class="btn btn-primary"
    >
      Next: Find Your Committee â†’
    </button>
  </div>
  <div
    style="margin-top: 2rem; padding: 1rem; background: rgb(248, 249, 250); border: 1px solid rgb(233, 236, 239); border-radius: 6px; font-size: 14px; color: rgb(108, 117, 125);"
  >
    <h3>
      ðŸ”„ What Happens Next
    </h3>
    <ul
      style="margin: 0.5rem 0px 0px 0px; padding-left: 1.2rem;"
    >
      <li>
        <strong>
          Find Your FEC Committee:
        </strong>
         We'll search for your registered committee
      </li>
      <li>
        <strong>
          Connect Bank Account:
        </strong>
         Secure connection via Plaid
      </li>
      <li>
        <strong>
          Launch Your Form:
        </strong>
         Get your embed code and start collecting
      </li>
    </ul>
    <h3
      style="margin-top: 1rem;"
    >
      ðŸ”’ Privacy & Security
    </h3>
    <ul
      style="margin: 0.5rem 0px 0px 0px; padding-left: 1.2rem;"
    >
      <li>
        <strong>
          Your information is encrypted
        </strong>
         and stored securely
      </li>
      <li>
        <strong>
          FEC-compliant data handling
        </strong>
         with audit trails
      </li>
      <li>
        <strong>
          You control your campaign data
        </strong>
         - export anytime
      </li>
    </ul>
    <h3
      style="margin-top: 1rem;"
    >
      ðŸ“‹ Requirements
    </h3>
    <ul
      style="margin: 0.5rem 0px 0px 0px; padding-left: 1.2rem;"
    >
      <li>
        <strong>
          Valid FEC committee
        </strong>
         registration required
      </li>
      <li>
        <strong>
          Campaign bank account
        </strong>
         for processing donations
      </li>
      <li>
        <strong>
          Official campaign website
        </strong>
         (recommended)
      </li>
    </ul>
    <p
      style="margin-top: 1rem; text-align: center; color: rgb(73, 80, 87);"
    >
      <strong>
        We'll guide you through the complete setup process.
      </strong>
      <br />
      This process takes about 5-10 minutes to complete.
    </p>
  </div>
</div>
`;
